packages/wagmi/src/client.ts

```diff
@@ -332,41 +320,47 @@ export class Web3Modal extends Web3ModalScaffold {
   private syncConnectors(wagmiConfig: Web3ModalClientOptions['wagmiConfig']) {
     const w3mConnectors: Connector[] = []
     wagmiConfig.connectors.forEach(({ id, name }) => {
-      if (id !== EIP6963_CONNECTOR_ID) {
+      if (id !== ConstantsUtil.EIP6963_CONNECTOR_ID) {
         w3mConnectors.push({
           id,
-          explorerId: ConnectorExplorerIds[id],
-          imageId: ConnectorImageIds[id],
+          explorerId: PresetsUtil.ConnectorExplorerIds[id],
+          imageId: PresetsUtil.ConnectorImageIds[id],
           imageUrl: this.options?.connectorImages?.[id],
-          name: ConnectorNamesMap[id] ?? name,
-          type: ConnectorTypesMap[id] ?? 'EXTERNAL'
+          name: PresetsUtil.ConnectorNamesMap[id] ?? name,
+          type: PresetsUtil.ConnectorTypesMap[id] ?? 'EXTERNAL'
         })
       }
     })
     this.setConnectors(w3mConnectors)
   }
 
+  private eip6963EventHandler(connector: EIP6963Connector, event: CustomEventInit<Wallet>) {
+    if (event.detail) {
+      const { info, provider } = event.detail
+      this.addConnector({
+        id: ConstantsUtil.EIP6963_CONNECTOR_ID,
+        type: 'ANNOUNCED',
+        imageUrl: info.icon ?? this.options?.connectorImages?.[ConstantsUtil.EIP6963_CONNECTOR_ID],
+        name: info.name,
+        provider,
+        info
+      })
+      connector.isAuthorized({ info, provider })
+    }
+  }
+
   private listenConnectors(wagmiConfig: Web3ModalClientOptions['wagmiConfig']) {
     const connector = wagmiConfig.connectors.find(
-      c => c.id === EIP6963_CONNECTOR_ID
+      c => c.id === ConstantsUtil.EIP6963_CONNECTOR_ID
     ) as EIP6963Connector
 
     if (typeof window !== 'undefined' && connector) {
-      window.addEventListener(EIP6963_ANNOUNCE_EVENT, (event: CustomEventInit<Wallet>) => {
-        if (event.detail) {
-          const { info, provider } = event.detail
-          this.addConnector({
-            id: EIP6963_CONNECTOR_ID,
-            type: 'ANNOUNCED',
-            imageUrl: info.icon ?? this.options?.connectorImages?.[EIP6963_CONNECTOR_ID],
-            name: info.name,
-            provider,
-            info
-          })
-          connector.isAuthorized({ info, provider })
-        }
-      })
-      window.dispatchEvent(new Event(EIP6963_REQUEST_EVENT))
+      const handler = this.eip6963EventHandler.bind(this, connector)
+      window.addEventListener(ConstantsUtil.EIP6963_ANNOUNCE_EVENT, handler)
+      window.dispatchEvent(new Event(ConstantsUtil.EIP6963_REQUEST_EVENT))
+      setTimeout(() => {
+        window.removeEventListener(ConstantsUtil.EIP6963_ANNOUNCE_EVENT, handler)
```

Comment:
Why are we removing listener here after 100ms? It should be active throughout whole app session. Per eip6963 spec there is no timeframe within which wallets should announce themselves, so we should listen all the time and filter out duplicate events if any happen.